File: 169-majority-element/majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int n = nums.length/2;
        Arrays.sort(nums);

        for(int i = 0; i <= n;i++){
            if(nums[i] == nums[n]){
                return nums[i];
            }
        }
        return 0;
    }
}

File: 169-majority-element/majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int n = nums.size();
        unordered_map<int, int> mpp;
        // int max_Apperance  = (n/2) + 1;
        // int count  = 0;

        for(int i = 0 ;  i < n ; i++  ){
           mpp[nums[i]]++;
        }

        for(auto it : mpp){
            int element  = it.first;
            int maj_element = it.second;

            if(maj_element > n/2){
                return it.first;
            }
        }
       return {};
       
    }
};